### ISSUE TITLE
type(Contract) parser error

### ISSUE THREAD

**Issue by 0age** [protofire/solhint#162]
Consider the following Solidity contract:
```
pragma solidity 0.5.11;


contract BadLint {
  function example() public {
    string memory problematic = type(BadLint).name;
  }
}
```

Running solhint on this contract results in the following error:
```
line 6:32 extraneous input 'type' expecting {'~', 'from', '(', '[', 'address', 'var', 'bool', 'string', 'byte', '++', '--', 'new', '+', '-', 'after', 'delete', '!', Int, Uint, Byte, Fixed, Ufixed, BooleanLiteral, DecimalNumber, HexNumber, HexLiteral, Identifier, StringLiteral}
```

Similar errors are raised when using `type(BadLint).creationCode` and `type(BadLint).runtimeCode`.

**Comment by fvictorio** [comment]
This should be fixed in 3.0. Please re-open if that's not the case. And sorry for taking so long!
